linera-views/src/backends/rocks_db.rs

```diff
@@ -193,55 +196,41 @@ impl RocksDbStoreExecutor {
         Ok(key_values)
     }
 
-    fn write_batch_internal(&self, mut batch: Batch) -> Result<(), RocksDbStoreInternalError> {
-        // NOTE: The delete_range functionality of RocksDB needs to have an upper bound in order to work.
-        // Thus in order to have the system working, we need to handle the unlikely case of having to
-        // delete a key starting with [255, ...., 255]
-        let len = batch.operations.len();
-        let mut keys = Vec::new();
-        for i in 0..len {
-            let op = batch.operations.get(i).unwrap();
-            if let WriteOperation::DeletePrefix { key_prefix } = op {
-                if get_upper_bound(key_prefix) == Bound::Unbounded {
-                    for short_key in self.find_keys_by_prefix_internal(key_prefix.to_vec())? {
-                        let mut full_key = self.root_key.clone();
-                        full_key.extend(key_prefix);
-                        full_key.extend(short_key);
-                        keys.push(full_key);
-                    }
-                }
-            }
-        }
```

Comment:
Does this change the semantics of a batch?

In general, if we have a `DeletePrefix` in a batch, and later a `Put` in that prefix, I guess we _do_ want that to be written, right? (So the old code violated that.)

Reply:
Yes, the semantics changes. I added a test that is passed by the new code and failed by the old.